File: Solutions/0795/0795.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 0795.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-11-24 11:43:01
***************************************************************************************************/

int numSubarrayBoundedMax(int* nums, int numsSize, int left, int right){
    int res=0,l=-1,lst=0;
    for (int i=0;i<numsSize;i++) {
        if (nums[i]>right) {
            l = i;
            lst=0;
        } else if (nums[i]>=left) {
            lst = i-l;
        }
        res += lst;
    }
    return res;
}

File: Solutions/0795/0795.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/11/24 11:32
# File: 0795.py
# Desc: 

class Solution:
    def numSubarrayBoundedMax(self, nums: List[int], left: int, right: int) -> int:
        l = []
        temp = []
        ma = 0
        for i in nums:
            if i > right:
                if ma >= left and temp:
                    l.append(temp)
                    ma = 0
                temp = []
            else:
                if i > ma: ma = i
                temp.append(i)
        if ma >= left and temp: l.append(temp)
        ret = 0
        for l2 in l:
            num = (len(l2)+1)*len(l2)//2
            temp = 0
            for i in l2:
                if i < left: temp += 1
                else:
                    if temp:
                        num -= (temp+1)*temp//2
                        temp = 0
            if temp: num -= (temp+1)*temp//2
            ret += num
        return ret
